include/envoy/secret/secret_callbacks.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * Callbacks invoked by a secret manager.
+ */
+class SecretCallbacks {
+public:
+  virtual ~SecretCallbacks() {}
+
+  virtual void onAddOrUpdateSecret() PURE;
```

Comment:
Do we have any future plan to add more method to this function? If no I think you can use `std::function` instead of the interface.

Reply:
no, I don't think we need to add more method for update callback.
But if we use std::function<void()>,   the problem is remove.  When the socket_factory is gone, the function has to be removed. how do we remove a std::function?  unless each function associates with a unique name.  It become too complicated.